### PR TITLE
Add make gen-crd-clients to the CI suite

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation

/kind feature

> /kind hotfix

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: build/Makefile [hunk 1]
```diff
@@ -842,3 +842,10 @@ update-go-deps:
 	$(DOCKER_RUN) go get -u agones.dev/agones
 	$(DOCKER_RUN) go mod tidy
 	$(DOCKER_RUN) go mod vendor
+
+test-gen-crd-client:
+    mkdir -p build/tmp
+    mv pkg/client build/tmp
+    make gen-crd-client
+    diff -r build/tmp/client pkg/client || true
```
Comment (gongmax) [inline build/Makefile:850]: The indent looks wrong. Each command should be after a `Tab`.

File: build/Makefile [hunk 2]
```diff
@@ -842,3 +842,10 @@ update-go-deps:
 	$(DOCKER_RUN) go get -u agones.dev/agones
 	$(DOCKER_RUN) go mod tidy
 	$(DOCKER_RUN) go mod vendor
+
+test-gen-crd-client:
+	mkdir -p build/tmp
+	mv ../pkg/client build/tmp
+	make gen-crd-client
+	diff -r build/tmp/client ../pkg/client || true
```
Comment (markmandel) [inline build/Makefile:850]: We do want this to fail , so let's remove the || true

For example: https://github.com/googleforgames/agones/blob/02365d2e0ea3b7605824a11f4980aec1a616c153/build/includes/website.mk#L111

(Although we don't need to sort, I don't think, but if this consistently fails, let's see what other approach)

Comment (markmandel) [reply build/Makefile:850]: As another though, is it worth deleting `build/tmp` if the diff passes?

File: cloudbuild.yaml [hunk 3]
```diff
@@ -115,6 +115,17 @@ steps:
...
   args: ["ensure-build-sdk-image-base"]
 
+#
+# Preventing Broken PR Merges in CI
+#
+- name: "make-docker"
+  id: test-gen-crd-client
+  waitFor:
+    - pull-build-image
+    - pull-build-sdk-base-image
```
Comment (markmandel) [inline cloudbuild.yaml:125]:
```suggestion
```

I don't think we need the sdk-base-image at this stage, so we can likely drop this.

File: build/Makefile (hunk 2 above)
Comment (Kalaiselvi84) [reply build/Makefile]: The code has been modified as per your suggestion.

File: cloudbuild.yaml (hunk 3 above)
Comment (Kalaiselvi84) [reply cloudbuild.yaml]: I've removed this line.

**Review by markmandel — APPROVED** [review]

**Pull request merged**